### PR TITLE
Add main ping

### PR DESCRIPTION
**Pull request by fbertsch**
Changes available here: https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/test-generated-schemas

Closes https://github.com/mozilla/mozilla-schema-generator/issues/43

### CODE REVIEW

File: mozilla_schema_generator/__main__.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def write_schema(repo, repo_id, config, out_dir, split, pretty, generic_schema):
     dump_schema(schemas, out_dir.joinpath(repo_id), pretty)
 
 
-def dump_schema(schemas, out_dir, pretty):
+def dump_schema(schemas, out_dir, pretty, *, version=1):
```
Comment (acmiyaguchi) [inline mozilla_schema_generator/__main__.py]: I haven't seen `*` used like this before, this is the same pattern as `*args, **kwargs`? Why not omit the `*`?

File: mozilla_schema_generator/main_ping.py [hunk 2]
```diff
@@ -8,64 +8,25 @@
...
-        schema._delete_key(("properties", "payload", "properties", "log"))
-
-        # 6. Update slices
-        for p in ("parent", "content", "gpu"):
-            schema.set_schema_elem(("properties", "payload", "properties", "processes", "properties", p, "properties", "gc", "properties", "random", "items", "properties", "slices", "type"), "number") # noqa E501
-            schema.set_schema_elem(("properties", "payload", "properties", "processes", "properties", p, "properties", "gc", "properties", "worst", "items", "properties", "slices", "type"), "number") # noqa E501
-
-        # 7. Ignore threadHangStats, deprecated in 57
-        schema._delete_key(("properties", "payload", "properties", "threadHangStats"))
-
```
Comment (acmiyaguchi) [inline mozilla_schema_generator/main_ping.py]: Nice.

File: mozilla_schema_generator/main_ping.py [hunk 3]
```diff
@@ -8,64 +8,25 @@
...
         return self._update_env(schema)
 
     def _update_env(self, schema):
-        # 2. Make partnerNames just a str array
-        schema.set_schema_elem(("properties", "environment", "properties", "partner", "properties", "partnerNames", "type"), "array") # noqa E501
-        schema.set_schema_elem(("properties", "environment", "properties", "partner", "properties", "partnerNames", "items"), {"type": "string"}) # noqa E501
+        schema._delete_key(prepend_properties(("environment", "settings", "userPrefs")))
```
Comment (acmiyaguchi) [inline mozilla_schema_generator/main_ping.py]: It's worth adding a note about why we're not adding `userPrefs` to the schema.

File: mozilla_schema_generator/__main__.py (hunk 1 above)
Comment (robhudson) [reply mozilla_schema_generator/__main__.py]: This is new to me too so I looked it up. [PEP 3102](https://www.python.org/dev/peps/pep-3102/) describes it in detail. Basically it forces anything after the `*` to be a keyword parameter and doesn't allow any more positional args aside from the ones defined like `*args` would do.

File: mozilla_schema_generator/main_ping.py (hunk 3 above)
Comment (jklukas) [reply mozilla_schema_generator/main_ping.py]: Does this line address #43?

Comment (acmiyaguchi) [reply mozilla_schema_generator/main_ping.py]: It does. Cases 4, 5, and 7 in the previous revision use the `_delete_key` method and the `prepend_properties` function just interleaves every other element in the list with a "property" key for traversing the JSON Schema.

**Pull request merged**